apis/tempo/v1alpha1/tempostack_types.go

```diff
@@ -431,10 +458,9 @@ type ObjectStorageSpec struct {
 	// Name of a secret in the same namespace as the tempo TempoStack custom resource.
 	//
 	// +kubebuilder:validation:Required
-	// +kubebuilder:validation:MinLength=1
 	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Object Storage Secret"
-	Secret string `json:"secret"`
-	// Don't forget to update storageSecretField in tempostacks_controller.go if this field name changes.
+	Secret ObjectStorageSecretSpec `json:"secret"`
+	// Don't forget to update storageSecretField in tempostack_controller.go if this field name changes.
```

Comment:
Can you update the `storageSecretField` constant in `tempostack_controller.go`? It specifies which field should be indexed (and it'll reconcile all TempoStacks which use the same referenced secret). I'll create an additional e2e test for this (separate PR), afaics the current e2e test should have failed but apparently did not fail.